src/metabase/search/appdb/core.clj

```diff
@@ -78,7 +78,9 @@
 (defmethod search.engine/model-set :search.engine/fulltext
   [search-ctx]
   ;; We ignore any current models filter
-  (let [search-ctx (assoc search-ctx :models search.config/all-models)]
+  (let [unfiltered-context (assoc search-ctx :models search.config/all-models)
+        applicable-models  (search.filter/search-context->applicable-models unfiltered-context)
+        search-ctx         (assoc search-ctx :models applicable-models)]
```

Comment:
This song and dance around stripping the API model selection, then reapplying the implied model filters, is a bit arcane. We should just update the FE to send valid `:models` in the first place and then we can delete this whole hack.

Reply:
Absolutely! This is weird.

Maybe we should collect "what to change in the FE" somewhere.